### PR TITLE
add vctk_test for VCTK dataset based on #812 PR

### PR DESCRIPTION
**Pull request by JianwuXu**
add vctk_test for VCTK dataset based on #812 PR. The unit test is based on the changes from #812.

### CODE REVIEW

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    TempDirMixin,
+    TorchaudioTestCase,
+    get_whitenoise,
+    save_wav,
+    normalize_wav,
+)
+
+
+# Used to generate a unique utterance for each dummy audio file
+NUMBERS = [
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:15]: This is not a part of VCTK. You can use some of the actual utterances from the dataset.

File: test/torchaudio_unittest/datasets/datasets_test.py [hunk 2]
```diff
@@ -1,7 +1,3 @@
-from torchaudio.datasets.librispeech import LIBRISPEECH
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/datasets_test.py:1]: You can delete the whole file.

**Comment by mthrok** [comment]
Please run your test with `pytest test/torchaudio_unittest/datasets/vctk_test.py`

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 1 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py:15]: The updated VCTK from PR #812 uses .flac format. Should I use .wav instead?

File: test/torchaudio_unittest/datasets/datasets_test.py (hunk 2 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/datasets_test.py:1]: this line has already been deleted in the first commit.

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 1 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py]: Update vctk_test.py to use actual utterances from the dataset and mimic the original data structure. However, the dataset VCTK-Corpus-0.92.zip contains the update data format in .flac form. The original vctk.py before PR #812 uses .wav format. I suppose we should .flac format?

**Comment by JianwuXu** [comment]
pytest test/torchaudio_unittest/datasets/vctk_test.py should work with the update vctk.py from PR #812 .

File: test/torchaudio_unittest/datasets/datasets_test.py (hunk 2 above)
Comment (mthrok) [reply test/torchaudio_unittest/datasets/datasets_test.py:1]: Not sure what you mean by the first commit, but this file no longer contains any test (as planned), so please remove the file itself.

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+class TestVCTK(TempDirMixin, TorchaudioTestCase):
+    backend = 'default'
+
+    root_dir = None
+    samples = []
+
+    _ext_txt = ".txt"
+    _ext_audio = ".wav"
+    _folder_txt = "txt"
+    _folder_audio = "wav48_silence_trimmed"
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:39]: None of these need to be variable or class attributed. Can you just embed in `setUpClass` function?

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    def setUpClass(cls):
+        cls.root_dir = cls.get_base_temp_dir()
+        dataset_dir = os.path.join(cls.root_dir, vctk.FOLDER_IN_ARCHIVE)
+        os.makedirs(dataset_dir, exist_ok=True)
+        sample_rate = 96000  # 96 kHz
+        seed = 0
+
+        for speaker in range(225, 230):
+            speaker_id = 'p' + str(speaker)
+            audio_path = os.path.join(dataset_dir, cls._folder_audio, speaker_id)
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:51]: calling a directory path `audio_path` looks confusing. Can you change it to something like `audio_dir`?

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            os.makedirs(audio_path, exist_ok=True)
+
+            file_path = os.path.join(dataset_dir, cls._folder_txt, speaker_id)
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:54]: same here, calling a directory path `file_path` looks confusing.

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    _ext_audio = ".wav"
+    _folder_txt = "txt"
+    _folder_audio = "wav48_silence_trimmed"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root_dir = cls.get_base_temp_dir()
+        dataset_dir = os.path.join(cls.root_dir, vctk.FOLDER_IN_ARCHIVE)
+        os.makedirs(dataset_dir, exist_ok=True)
+        sample_rate = 96000  # 96 kHz
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:46]: This comment is redundant.

Comment (mthrok) [reply test/torchaudio_unittest/datasets/vctk_test.py:46]: Also did you find an audio file of 96 kHz? I did some spot checks and all of them are 48000 Hz.

```
soxi ~/VCTK-Corpus-0.92/wav48_silence_trimmed/p225/p225_001_mic*

Input File     : '/Users/moto/VCTK-Corpus-0.92/wav48_silence_trimmed/p225/p225_001_mic1.flac'
Channels       : 1
Sample Rate    : 48000
Precision      : 16-bit
Duration       : 00:00:02.05 = 98473 samples ~ 153.864 CDDA sectors
File Size      : 106k
Bit Rate       : 413k
Sample Encoding: 16-bit FLAC


Input File     : '/Users/moto/VCTK-Corpus-0.92/wav48_silence_trimmed/p225/p225_001_mic2.flac'
Channels       : 1
Sample Rate    : 48000
Precision      : 16-bit
Duration       : 00:00:02.05 = 98473 samples ~ 153.864 CDDA sectors
File Size      : 92.4k
Bit Rate       : 360k
Sample Encoding: 16-bit FLAC
```

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        for i, (data, sample_rate, utterance, speaker_id, utterance_id) in enumerate(dataset):
+            self.assertEqual(data, self.samples[i][0], atol=5e-5, rtol=1e-8)
+            assert sample_rate == self.samples[i][1]
+            assert utterance == self.samples[i][2]
+            assert speaker_id == self.samples[i][3]
+            assert int(utterance_id) == self.samples[i][4]
+            num_samples += 1
+
+        assert num_samples == len(self.samples)
+        vctk.VCTK._ext_audio = '.flac'
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:104]: You do not need to do this in test method as it is done in `tearDownClass`.

File: test/torchaudio_unittest/datasets/vctk_test.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+                seed += 1
+
+    @classmethod
+    def tearDownClass(cls):
+        # In case of test failure
+        vctk.VCTK._ext_audio = '.flac'
+
+    def test_vctk(self):
+        vctk.VCTK._ext_audio = '.wav'
```
Comment (mthrok) [inline test/torchaudio_unittest/datasets/vctk_test.py:92]: Can you move this to `setUpClass` method?

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 7 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py:104]: corrected

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 6 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py:46]: corrected

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 5 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py:54]: modified accordingly

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 4 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py:51]: modified accordingly

File: test/torchaudio_unittest/datasets/vctk_test.py (hunk 3 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/vctk_test.py:39]: modified accordingly

File: test/torchaudio_unittest/datasets/datasets_test.py (hunk 2 above)
Comment (JianwuXu) [reply test/torchaudio_unittest/datasets/datasets_test.py:1]: test/torchaudio_unittest/datasets/datasets_test.py removed

**Comment by Abhi011999** [comment]
@JianwuXu time to write tests for new `VCTK_092` class as work in #812 is already complete.

**Comment by JianwuXu** [comment]
@Abhi011999 thanks for your contribution regarding #812. I will proceed accordingly on the testing.

**Comment by JianwuXu** [comment]
@mthrok updated vctk_test.py. The unit test passed based on the latest of PR #812 for VCTK_092 class. However, since the file format '.flac' is hard-coded in VCTK_092 class, I don't think vctk_test.py is able to change that and have to use '.flac' for testing. Let me know what you think.

**Comment by mthrok** [comment]
> @mthrok updated vctk_test.py. The unit test passed based on the latest of PR #812 for VCTK_092 class. However, since the file format '.flac' is hard-coded in VCTK_092 class, I don't think vctk_test.py is able to change that and have to use '.flac' for testing. Let me know what you think.

Hi @JianwuXu 

I tweaked VCTK class in https://github.com/mthrok/audio/commit/6f11a4f82f5c6a5822c4a8e552e603cf9239e953, which gives `audio_ext` argument. Can you cherry-pick that one and rebase your work on top of it??

**Comment by JianwuXu** [comment]
Hi @mthrok, I have rebased PR #875  to mthrok@6f11a4f and use '.wav' as testing format. It passed the test 'pytest test/torchaudio_unittest/datasets/vctk_test.py -s'. let me know if you have any comments.

**Pull request merged**